Create guide for xml to json transformation (#242)
triggermesh/docs: docs/guides/xmltojsontransformation.md

# Transforming XML to JSON

The TriggerMesh `XMLToJSONTransformation` API object can be used to process a Cloudevent containing XML and return a JSON representation.

![](./../assets/images/xmltojsontransformation.png)

## Configuring an XML to JSON event flow XML

> This guide shows you how to configure an event flow which transforms incoming CloudEvent in XML to their JSON representation. It has four steps:

Reviewer: ```suggestion
This guide shows you how to configure an event flow that transforms an incoming CloudEvent in XML to their JSON representation. It has four steps:
```

* Deploy the `EventDisplay` service.

> * Deploy the `XMlToJSONTransformation`.

Reviewer: ```suggestion
* Deploy the `XMlToJSONTransformation` object.
```

* Deploy a Source that emits XML data.

> * Check the results in the logs of `EventDisplay` Pod. 

Reviewer: ```suggestion
* Check the results in the logs of the `EventDisplay` Pod. 
```

> A `XMLToJSONTransformation` can be configured to either reply to the event sender or to send the transformed data to a `Sink` if one is provided. In this guide, we will use a `Sink` to send the transformed data to a so-called `EventDisplay` service. 

Reviewer: ```suggestion
An `XMLToJSONTransformation` object can be configured to either reply to the event sender or to send the transformed data to a `Sink` if one is provided. In this guide, we will use a `Sink` to send the transformed data to a so-called `EventDisplay` service. 
```

### Deploying an Event Display
Let's first deploy the end of our event flow. The `EventDisplay` is a simple application that can be used to display CloudEvents. It can 
be deployed by writing the following YAML in a file and using `kubectl apply -f <manifest.yaml>`:

```yaml
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
    name: event-display
spec:
  template:
    spec:
      containers:
        - image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display@sha256:46d5a66f300c3ced590835d379a0e9badf413ae7ab60f21a2550ecedbc9eb9d3
```

### Deploying an XMLToJSONTransformation Object
With the `event-display` in place, the `XMLToJSONTransformation` object can now be deployed in the same manner using the following manifest:

!!! Tip
    Below we use a `Sink` to declare where the response go. If you omit the `Sink` the response will go back to the Sender.

```yaml
apiVersion: flow.triggermesh.io/v1alpha1
kind: XMLToJSONTransformation
metadata:
  name: demo
spec:
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: event-display
```

### Deploying a `PingSource` Object.

> Finally, we deploy an event source that emits CloudEvent with XML. We can do this with the `PingSource` which sends Cloudevents on a 

Reviewer: ```suggestion
Finally, we deploy an event source that will emit CloudEvents with XML data in the payload. We can do this with the `PingSource` which sends Cloudevents on a 
```
